src/doppkit_toolbox/esri/toolboxes/doppkit_wrapper_toolbox.pyt

```diff
@@ -77,26 +87,49 @@ class FetchExport:
             parameterType="Required",
             direction="Input",
         )
+
         add_to_map.value = True
+
         return [grid_server, grid_access_token, aoi_name, dl_directory, add_to_map]
 
     def updateParameters(self, parameters):
         """Modify the values and properties of parameters before internal
         validation is performed.  This method is called whenever a parameter
         has been changed."""
+        #     if parameters[0].altered:
+        #         # thank you @fatih_dur https://gis.stackexchange.com/a/294476
+        #         new_values = [i[0] for i in parameters[0].values if i[0] not in parameters[0].filters[0].list]
+        #         parameters[0].filters[0].list += new_values
         return
 
     def updateMessages(self, parameters):
         """Modify the messages created by internal validation for each tool
         parameter.  This method is called after internal validation."""
+        if parameters[0].value not in [
+            "https://grid.nga.mil/grid",
+            "https://grid.nga.smil.mil",
+            "https://grid.nga.ic.gov",
+        ]:
+            parameters[0].clearMessage()
         return
 
     def execute(self, parameters, messages):
         """The source code of the tool."""
 
-        named_parameters = SyncParameters(
-            *parameters
-        )
+        arcpy.AddMessage("====== PARAMETERS ======")
+        for parameter_print in parameters:
+            if parameter_print.displayName in [
```

Comment:
consider not checking by displayName but by another attribute that is not likely to change as much.